chapters/operatorsandexpressions.tex

```diff
@@ -103,10 +103,9 @@ \section{Operator Precedence and Associativity}\label{operator-precedence-and-as
 \end{center}
 \end{table}
 
-The postfix array index and postfix access operators are not merely expression operators in the normal sense that \lstinline!a.b[1]! can be treated as \lstinline!a.(b[1])!.
-Instead, these operators need to be considered jointly to identify an entire \lstinline[language=grammar]!component-reference! (one of the alternative productions for \lstinline[language=grammar]!primary! in the grammar) which is the smallest unit that can be seen as an expression in itself.
-Postfix access can only be applied immediately to a \lstinline[language=grammar]!component-reference!; not even parentheses around the left operand are allowed.
-Postfix array index can additionally be applied if there are parentheses around the left operand, see \cref{indexing}.
+The postfix array index and postfix access operators can both be part of a \lstinline[language=grammar]!component-reference! (one of the alternative productions for \lstinline[language=grammar]!primary! in the grammar) and be applied to general expressions when the left operand is parenthesized.
+Directly using both postfix access operator and postfix array index in a \lstinline[language=grammar]!component-reference! has a special intuitive meaning.
+See \cref{indexing} and \cref{member-access-operator}.
```

Comment:
I think we can leave the intuitiveness for the reader to decide:
```suggestion
There are semantic differences between the \lstinline[language=grammar]!component-reference! forms and the parenthesized forms of the operators, see \cref{indexing} and \cref{member-access-operator}.
```

Reply:
```suggestion
Directly using both postfix access operator and postfix array index in a \lstinline[language=grammar]!component-reference! has a special meaning.
See \cref{indexing} and \cref{member-access-operator}.
```
I would then propose to just remove "intuitive" as above.

The key point is that only the case where you combine the two operators has a special meaning for component-references, and that is not syntactically legal for the parenthesized forms.

Reply:
Sounds good.